tests/php/CampaignAdminTest.php

```diff
@@ -129,4 +134,72 @@ public function testFilters()
             $results
         );
     }
+
+    public function testReadCampaignResponse()
+    {
+        // Login as admin user
+        $group = $this->objFromFixture(Group::class, 'admins');
+        Permission::grant($group->ID, 'ADMIN');
+
+        $admin = $this->objFromFixture(Member::class, 'admin_user');
+        $this->logInAs($admin);
+
+        // Create new ChangeSet object
+        $changeset = ChangeSet::create();
+        $changeset->ID = '123';
+        $changeset->Name = 'changeset 123';
+        $changeset->State = 'open';
+        $changeset->IsInferred = false;
+        
+        $changeset->write();
+
+        // ChangeSet should be accessable for Admin
+        $this->assertTrue($changeset->canView());
+
+        $expectedStatus = [
+            '200' => [
+                'ID' => '123',
+                'Name' => 'show'
+            ],
+            '400' => [
+                'ID' => '123',
+                'Name' => ''
+            ],
+            '404' => [
+                'ID' => '12345',
+                'Name' => 'show'
+            ]
+        ];
+
+        // Create new CampaignAdmin object
+        $campaignAdmin = CampaignAdmin::create();
+
+        $request = new HTTPRequest('GET', '/admin/campaigns/set/');
+        $request->addHeader('Accept', 'application/json');
+
+        foreach ($expectedStatus as $key => $val) {
+            $request->setRouteParams($val);
+            $response = $campaignAdmin->readCampaign($request);
+            $status = $response->getStatusCode();
+
+            // Method should return correct status
+            $this->assertInstanceOf(HTTPResponse::class, $response);
+            $this->assertEquals($key, $status);
+        }
+
+        // Login as user without admin permissions
+        $user = $this->objFromFixture(Member::class, 'mock_user');
```

Comment:
That's a good idea, but that's not really a valid test. A user without any permission at all, won't be bale to make it to you method in the first place, because the controller wouldn't relay your request to the action.

Reply:
DONE